daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/BinaryNumberTraits.scala

```diff
@@ -108,15 +109,32 @@ trait PrefixedLengthParserMixin {
    */
   def getPrefixedLengthInBits(state: PState): Long = {
     val lenInUnits = getPrefixedLengthInUnits(state)
-    lengthUnits match {
+    val lenInBits = lengthUnits match {
       case LengthUnits.Bits => lenInUnits
       case LengthUnits.Bytes => lenInUnits * 8
       case LengthUnits.Characters => {
         val mfw = state.encoder.bitsCharset.maybeFixedWidth
-        if (mfw.isDefined) mfw.get
+        if (mfw.isDefined)
+          lenInUnits * mfw.get // fixed, only supported for parsing (maybe)
         else
           Assert.invariantFailed("Prefixed length for text data in non-fixed width encoding.")
       }
     }
+    // Now we save the contentLength that we obtained from the prefix
+    // so that the dfdl:contentLength function can be called on the prefixed length element.
+    // as parse time.
```

Comment:
Typo in the comment here.  Assuming you don't want the period on the previous line and "at parse time" instead of "as parse time".